patches/tModLoader/Terraria/ModLoader/ModAccessorySlot.cs

```diff
@@ -0,0 +1,83 @@
+using Microsoft.Xna.Framework;
+using System.Collections.Generic;
+using Terraria.UI;
+
+//NOTE: Does not fully support: ItemLoader.GetGamepadInstructions and related gamepad stuff.
+
+namespace Terraria.ModLoader
+{
+	/// <summary>
+	/// A ModAccessorySlot instance represents a net new accessory slot instance. You can store fields in the ModAccessorySlot class. 
+	/// </summary>
+	public abstract class ModAccessorySlot : ModType
+	{
+		internal virtual bool skipRegister => false;
+
+		internal int slot;
+		internal int index;
+
+		// Setting toggle for stack or scroll accessories/npcHousing
+		internal static bool scrollSlots = true;
+		internal static int scrollbarSlotPosition = 0;
+
+		// List of registered identifiers for modded accessory slots. Used in DefaultPlayer.
+		internal static List<string> moddedAccSlots = new List<string>();
+
+		/// <summary>
+		/// Called when the ModAccessorySlot is registered. Registers to parent list.
+		/// Requests for a modded Slot, and stores assigned Slot. DefaultPlayer will auto-manage the items themselves.
+		/// </summary>
+		internal protected void Initialize() {
```

Comment:
`private`?
`internal` allows accesses from tML outside the `ModAccessorySlot` class
`protected` allows accesses from subclasses

Just merge this into `Register`